app/CreateDeal.tsx

```diff
@@ -84,13 +147,38 @@ export function CreateDeal(props: {
             if (pricePerUnit === "0") {
               return setErrorMessage("Price cannot be $0");
             }
-
-            console.log(
-              "item:" + itemName,
-              typeof itemName,
-              "price per unit:" + pricePerUnit,
-              typeof pricePerUnit
+            if (selectedUnit === undefined) {
+              return setErrorMessage("Must select a unit");
+            }
+            if (selectedStoreId === undefined) {
+              return setErrorMessage("Must select a store");
+            }
+            if (selectedDate === undefined) {
+              return setErrorMessage("Must select a date");
+            }
+            setErrorMessage("");
+            setAwaitingCreateDealCheck(true);
+            if (
+              await itemAlreadyExists(
+                itemName,
+                Number(pricePerUnit),
+                selectedStoreId
+              )
+            ) {
+              setAwaitingCreateDealCheck(false);
+              return setErrorMessage(
+                "An item with the same details already exists"
+              );
+            }
+            onCreateDealClick();
```

Comment:
what does `onCreateDealClick` do? You aren't always calling this when create deal is clicked so probably a bad function name